app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,62 @@
+class AlbumsController < ApplicationController
+  before_action :get_album, only: [:edit, :update, :destroy]
+  def get_album
+    @album = Album.find(params[:id])
+  end
+
+  def index
+    @albums = Album.all
+  end
+
+  def new
+    @url = "create"
+    @legend = "New Album"
+    @album = Album.new
+  end
+
+  def create
+    @album = Album.create(album_params[:album])
+    if @album.save
+      redirect_to albums_path
+    else
+      render :new
+    end
+  end
+
+  def show
+    @album = Album.find(params[:id])
```

Comment:
You made that handy dandy get_album before method but are re-writing that code here.  Also you could add upvote and edit to your before action as those are also using ```@album = Album.find(params[:id])```